File: DarkOrbit 10.0/Net/netty/commands/BattleStationBuildingUiInitializationCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class BattleStationBuildingUiInitializationCommand
    {
        public const short ID = 770;

        public static byte[] write(int mapAssetId, int battleStationId, string battleStationName, AsteroidProgressCommand progress, AvailableModulesCommand availableModules, int buildTimeInMinutesMin, int buildTimeInMinutesMax, int buildTimeInMinutesIncrement)
        {
            var param1 = new ByteArray(ID);
            param1.write(progress.write());
            param1.writeInt(buildTimeInMinutesMax << 4 | buildTimeInMinutesMax >> 28);
            param1.writeUTF(battleStationName);
            param1.writeInt(mapAssetId << 14 | mapAssetId >> 18);
            param1.writeInt(buildTimeInMinutesMin >> 14 | buildTimeInMinutesMin << 18);
            param1.writeInt(buildTimeInMinutesIncrement << 14 | buildTimeInMinutesIncrement >> 18);
            param1.writeInt(battleStationId << 3 | battleStationId >> 29);
            param1.writeShort(-7693);
            param1.write(availableModules.write());
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/BattleStationBuildingUiInitializationCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 22-03-2015.
 */
public class BattleStationBuildingUiInitializationCommand
        implements ServerCommand {

    public static int ID = 28731;

    public int                     buildTimeInMinutesIncrement;
    public AvailableModulesCommand availableModules;
    public int                     battleStationId;
    public int                     buildTimeInMinutesMax;
    public String                  battleStationName;
    public AsteroidProgressCommand progress;
    public int                     buildTimeInMinutesMin;
    public int                     mapAssetId;

    public BattleStationBuildingUiInitializationCommand(int param1, int param2, String param3,
                                                        AsteroidProgressCommand param4, AvailableModulesCommand param5,
                                                        int param6, int param7, int param8) {
        this.mapAssetId = param1;
        this.battleStationId = param2;
        this.battleStationName = param3;
        this.progress = param4;
        this.availableModules = param5;
        this.buildTimeInMinutesMin = param6;
        this.buildTimeInMinutesMax = param7;
        this.buildTimeInMinutesIncrement = param8;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.buildTimeInMinutesIncrement >>> 15 | this.buildTimeInMinutesIncrement << 17);
            this.availableModules.write(param1);
            param1.writeInt(this.battleStationId << 15 | this.battleStationId >>> 17);
            param1.writeInt(this.buildTimeInMinutesMax << 13 | this.buildTimeInMinutesMax >>> 19);
            param1.writeUTF(this.battleStationName);
            param1.writeShort(9415);
            this.progress.write(param1);
            param1.writeInt(this.buildTimeInMinutesMin >>> 15 | this.buildTimeInMinutesMin << 17);
            param1.writeInt(this.mapAssetId << 10 | this.mapAssetId >>> 22);
        } catch (IOException e) {
        }
    }
}
